### ISSUE TITLE
Scroll to first unread message

### ISSUE THREAD

**Issue by jurchiks** [gitterHQ/gitter#478]
There's this annoying thing in the current chat system that whenever you've been away for the night and come back in the morning, if there have been discussions happening in the chat during that time, there is no way to just scroll to the first unread message since I left.
I find that annoying as I have to scroll up and wait for the ajax to load every time I do so as it only loads a fixed amount of messages on every request.

What I'd like to see when I open a chat room that has unread messages is a button, clicking which would put me at the last few read messages (for context) and start loading the unread messages from the bottom, as usual. That would basically be the same as if I had just walked away from the PC and left the chat open slightly scrolled up, since, as far as I've noticed, the chat keeps scrolling when new messages arrive even when there are many of them and none are marked read. It would be better to just show the "x new messages" bar at the bottom when new messages arrive, and the reader has the option to scroll down to read them. Automatic scrolling makes this an annoying experience. At the very least add a setting to disable it.

TL;DR:
1) option to disable auto-scrolling for new messages (mirc has this done pretty well - it only scrolls if you're focused on the window, if you're not, it just loads the new messages below the visible area, and even then it marks the last known read message with a red line below it, just in case you're afk);
2) option to scroll up to first unread message after being away for a while (i.e. computer turned off).

**Comment by agauniyal** [comment]
:+1: same experience for me.

**Comment by mydigitalself** [comment]
Don't you see a green banner at the top of the chat? You can click on that and it will navigate to the first unread message.

**Comment by damianavila** [comment]
>Don't you see a green banner at the top of the chat? You can click on that and it will navigate to the first unread message.

Sometimes you have sequential navigation, I mean, you have to press the green banner multiple times to get to the first unread message... In any case, the scroll behaviour seems erratic very frequently...

**Comment by mydigitalself** [comment]
Hmmm, that's not right. We **used** to have a limit to the number it would jump to, but it should always be jumping now to the first unread item. When it next happens, could you perhaps let me know the number of unread messages it's not jumping to correctly and I'll see if I can try to reproduce. I just tested on a room with 47 unread messages and it worked perfectly.

What other sort of erratic behaviour do you experience?

**Comment by damianavila** [comment]
ok, I will try to record a video showing it my experience... I think it will be the best way to show you how it feels for me...

**Comment by mydigitalself** [comment]
Thanks, much appreciated!

**Comment by stefanv** [comment]
@damianavila Any update on this?

**Comment by damianavila** [comment]
@stefanv I did not have much time to test it... I will try to record the video this week... do you experience some of this issues too?

**Comment by stefanv** [comment]
@damianavila I came here looking for a way to scroll to the last read message, and I will report back if I see anything strange on my platform.

**Comment by jurchiks** [comment]
@mydigitalself try in a room with a few hundred unread messages. It doesn't always scroll correctly since there are multiple pages to load. For some reason, when it should just load the first unread messages and then load the following ones when scrolling down, what it does instead is load ALL the unread messages part by part. IMHO that's wrong.

**Comment by quintesse** [comment]
> Don't you see a green banner at the top of the chat? You can click on that and it will navigate to the first unread message.

@mydigitalself Ok I just found that out because you mentioned it here, but that's not at all intuitive, especially when there's a button to mark everything as read you don't expect the rest of the text to work like a button as well. I think this could be made much more obvious, maybe just an icon of a arrow pointing up (or the same hand icon that is used for references) included somewhere in that line would help.
